docs/create_new_backend.rst

```diff
@@ -0,0 +1,483 @@
+----------------------
+
+For each new backend library, you should create the following two header files:
+
+* Header file with a declaration of entry points to the new third-party library wrappers
+* Compiler-time dispatching interface (see `oneMKL Usage Models <../README.md#supported-usage-models>`_) for new third-party libraries
+
+**Header File Example**: command to generate the header file with a declaration of BLAS entry points in the onemkl::newlib namespace 
+
+.. code-block:: bash
+
+    python scripts/generate_backend_api.py include/onemkl/blas/blas.hpp \                             # Base header file
+                                           include/onemkl/blas/detail/newlib/onemkl_blas_newlib.hpp \ # Output header file
+                                           onemkl::newlib                                             # Wrappers namespace
+
+Code snippet of the generated header file ``include/onemkl/blas/detail/newlib/onemkl_blas_newlib.hpp``
+
+.. code-block:: cpp
+
+    namespace onemkl {
+    namespace newlib {
+    
+    void asum(cl::sycl::queue &queue, std::int64_t n, cl::sycl::buffer<float, 1> &x, std::int64_t incx,
+              cl::sycl::buffer<float, 1> &result);
+
+
+
+**Compile-time Dispatching Interface Example**: command to generate the compile-time dispatching interface template instantiations for ``newlib`` and supported device ``newdevice``
+
+.. code-block:: bash
+
+    python scripts/generate_ct_instant.py   include/onemkl/blas/detail/blas_ct_templates.hpp \         # Base header file
+                                            include/onemkl/blas/detail/newlib/blas_ct.hpp \            # Output header file
+                                            include/onemkl/blas/detail/newlib/onemkl_blas_newlib.hpp \ # Header file with declaration of entry points to wrappers
+                                            newlib \                                                   # Library name
+                                            newdevice \                                                # Backend name
+                                            onemkl::newlib                                             # Wrappers namespace
+
+Code snippet of the generated header file ``include/onemkl/blas/detail/newlib/blas_ct.hpp``
+
+.. code-block:: cpp
+
+    namespace onemkl {
+    namespace blas {
+    
+    template <>
+    void asum<library::newlib, backend::newdevice>(cl::sycl::queue &queue, std::int64_t n,
+                                                   cl::sycl::buffer<float, 1> &x, std::int64_t incx,
+                                                   cl::sycl::buffer<float, 1> &result) {
+        asum_precondition(queue, n, x, incx, result);
+        onemkl::newlib::asum(queue, n, x, incx, result);
+        asum_postcondition(queue, n, x, incx, result);
+    }
+
+
+.. _integrate_header_files:
+
+2. Integrate Header Files
+-------------------------
+
+Below you can see oneMKL Header-based Part Structure:
```

Comment:
"Below you can see the oneMKL directory structure for headers:" might be easier to understand.

Reply:
We also have headers in src/, so I tried to specify that this is structure inside include/. Do you think it's not really important?

Reply:
Ah, right :) Maybe "The diagram below shows the directory structure of the oneMKL external headers"?

Reply:
Sound better, I guess "detail" folders are not really external, so how about "Below you can see structure of oneMKL top-level include directory"?

Reply:
That sounds good to me, too.

Reply:
Thank you! I've pushed all suggested changes